### ISSUE TITLE
Reset-HPOVEnclosureDevice command failed

### ISSUE THREAD

**Issue by valfr** [HewlettPackard/POSH-HPOneView#239]
### Expected Behavior
Reset-HPOVEnclosureDevice -Efuse -Enclosure $myEnclosure -Component ICM -DeviceID 3
=> interconnect module #3 reset from $myEnclosure

### Actual Behavior
PS C:\Users\user> Reset-HPOVEnclosureDevice -Efuse -Enclosure $toto
cmdlet Reset-HPOVEnclosureDevice at command pipeline position 1
Supply values for the following parameters:
Component: ICM
DeviceID: 3
Confirm
Are you sure you want to perform this action?
Performing the operation "Reset power for device" on target "ICM 3 within
SNXXXXXX".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help
(default is "Y"):Y
Reset-HPOVEnclosureDevice : The "type" field content, "Replace", included in
this request is invalid. Correct the content in the 'type' field of the JSON
and try the request again.
At line:1 char:1
+ Reset-HPOVEnclosureDevice -Efuse -Enclosure $toto
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Send-HPOVRequest:String) [Res
   et-HPOVEnclosureDevice], InvalidOperationException
    + FullyQualifiedErrorId : InvalidAuthOperation,Reset-HPOVEnclosureDevice

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion`): 3.0.1221.2746 
HPE OneView Appliance Version:  Synergy 3.0

**Comment by ArnoutVerbeken** [comment]
Confirmed while setting up a new Synergy Frame. 

This is our output, using latest library on an up to date Synergy Frame:
`
```
PS C:\Windows\system32> Reset-HPOVEnclosureDevice -Enclosure BED2PXPSY01 -Component "Device" -DeviceID 2 -Efuse
You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.300\HPOneView.300.psm1:26266 char:5
+ ...             $ErrorRecord = New-ErrorRecord HPOneview.EnclosureResourc ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

Exception calling "ThrowTerminatingError" with "1" argument(s): "Value cannot be null.
Parameter name: errorRecord"
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.300\HPOneView.300.psm1:26267 char:5
+                 $PSCmdlet.ThrowTerminatingError($ErrorRecord)
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentNullException

You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.300\HPOneView.300.psm1:26355 char:4
+             $_Message = 'An invalid Enclosure object type was provide ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.300\HPOneView.300.psm1:26356 char:4
+             $ErrorRecord = New-ErrorRecord HPOneview.EnclosureResourc ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

Exception calling "ThrowTerminatingError" with "1" argument(s): "Value cannot be null.
Parameter name: errorRecord"
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.300\HPOneView.300.psm1:26357 char:4
+             $PSCmdlet.ThrowTerminatingError($ErrorRecord)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentNullException

You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.300\HPOneView.300.psm1:26365 char:4
+             $ErrorRecord = New-ErrorRecord HPOneview.EnclosureResourc ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

Exception calling "ThrowTerminatingError" with "1" argument(s): "Value cannot be null.
Parameter name: errorRecord"
At C:\Program Files\WindowsPowerShell\Modules\HPOneView.300\HPOneView.300.psm1:26366 char:4
+             $PSCmdlet.ThrowTerminatingError($ErrorRecord)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentNullException
```
`

We have put the enclosure in a variable, which gives us different output (still an error) but this should already work.

After using the $Enclosure variable, we get the following error:
`
```
Reset-HPOVEnclosureDevice -Enclosure $enclosure -Component
Device -DeviceID 2

Confirm
Are you sure you want to perform this action?
Performing the operation "Reset power for device" on target "Device 2 within
Encl1".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help
(default is "Y"):Y
Reset-HPOVEnclosureDevice : The "type" field content, "Replace", included in
this request is invalid. Correct the content in the 'type' field of the JSON
and try the request again.
At line:1 char:1
+ Reset-HPOVEnclosureDevice -Enclosure $enclosure -Component Device -DeviceID 2
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Send-HPOVRequest:String) [Res
   et-HPOVEnclosureDevice], InvalidOperationException
    + FullyQualifiedErrorId : InvalidAuthOperation,Reset-HPOVEnclosureDevice

```
`

**Comment by ChrisLynchHPE** [comment]
Yes, this is a known defect, and one I have fixed.  It will be included in the next 3.00 release.

If you need a workaround, let me know as I can provide a script example.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 3.0.1264.2772](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v3.0.1264.2772).

**Comment by ChrisLynchHPE** [comment]
Closing issue due to no further activity.  Feel free to re-open the issue if it still remains.
